src/mesh/meshblock.cpp

```diff
@@ -109,118 +117,13 @@ MeshBlock::MeshBlock(int igid, int ilid, LogicalLocation iloc, RegionSize input_
 
   // mesh-related objects
   // Boundary
-  pbval = std::make_unique<BoundaryValues>(this, input_bcs, pin);
+  pbval = std::make_unique<BoundaryValues>(shared_from_this(), input_bcs, pin);
   pbval->SetBoundaryFlags(boundary_flag);
 
   // Reconstruction: constructor may implicitly depend on Coordinates, and PPM variable
   // floors depend on EOS, but EOS isn't needed in Reconstruction constructor-> this is
   // ok
-  precon = std::make_unique<Reconstruction>(this, pin);
-
-  // Add field properties data
-  for (int i = 0; i < properties.size(); i++) {
-    StateDescriptor &state = properties[i]->State();
-    for (auto const &q : state.AllFields()) {
-      real_container->Add(q.first, q.second);
-    }
-    for (auto const &q : state.AllSparseFields()) {
-      for (auto const &m : q.second) {
-        real_container->Add(q.first, m);
-      }
-    }
-  }
-  // Add physics data
-  for (auto const &pkg : packages) {
-    for (auto const &q : pkg.second->AllFields()) {
-      real_container->Add(q.first, q.second);
-    }
-    for (auto const &q : pkg.second->AllSparseFields()) {
-      for (auto const &m : q.second) {
-        real_container->Add(q.first, m);
-      }
-    }
-  }
-
-  // TODO(jdolence): Should these loops be moved to Variable creation
-  ContainerIterator<Real> ci(real_container, {Metadata::Independent});
-  int nindependent = ci.vars.size();
-  for (int n = 0; n < nindependent; n++) {
-    RegisterMeshBlockData(ci.vars[n]);
-  }
-
-  if (pm->multilevel) {
-    pmr = std::make_unique<MeshRefinement>(this, pin);
-    // This is very redundant, I think, but necessary for now
-    for (int n = 0; n < nindependent; n++) {
-      pmr->AddToRefinement(ci.vars[n]->data, ci.vars[n]->coarse_s);
-    }
-  }
-
-  // Create user mesh data
```

Comment:
Maybe I'm missing something, but where does this happen now?

Reply:
It happens [here](https://github.com/lanl/parthenon/blob/a004fa7ac8f5f7964a62b0f68d6bebb0344105d6/src/mesh/meshblock.cpp#L126). There were three `MeshBlock` constructors, which all (as far as I can tell) were identical but slightly different signatures and duplicated code. So I combined them.

Reply:
Silly me... I just looked at the diff and didn't look at the whole file. Thanks for clarifying.